run-build-functions.sh

```diff
@@ -55,11 +55,18 @@ mkdir -p $NETLIFY_CACHE_DIR/.composer
 mkdir -p $NETLIFY_CACHE_DIR/.gimme_cache/gopath
 mkdir -p $NETLIFY_CACHE_DIR/.gimme_cache/gocache
 mkdir -p $NETLIFY_CACHE_DIR/.wasmer/cache
+mkdir -p $NETLIFY_CACHE_DIR/.cargo/registry
 
 : ${YARN_FLAGS=""}
 : ${NPM_FLAGS=""}
 : ${BUNDLER_FLAGS=""}
 
+check_rust_target_cacheability() {
+  # Check if repo/target is ignored by the repository. If it is, it's OK to
+  # cache it.
+  git --git-dir=${NETLIFY_REPO_DIR}/.git check-ignore -q ${NETLIFY_REPO_DIR}/target
+}
```

Comment:
@mraerino Thanks a lot for your help and review btw!! :heart: 

Do you know what is the purpose of this? It seems it checks if `target` is in the `.gitignore` and if that's the case it might not restore `target` from cache.

I think it's good to have a button to purge the cache but here it seems to use a git feature (gitignore) to detect wether or not target should be cached. It doesn't sound right to me and I see no reason why you would do that.

Reply:
In the end I removed it. It makes no sense.